### ISSUE TITLE
Multipass launch failed

### ISSUE THREAD

**Issue by ghost** [canonical/multipass#2620]
**Describe the bug**
Describe what your problem is.

I'm trying to build an "snap" on a RaspberryPi 3, OS ubuntu 20.04.4 LTS (GNU/Linux 5.4.0-1065-raspi armv7l)

1. For doing that I have installed snapcraft first:
  sudo snap install snapcraft --classic

2. in my home directory, I generate a snapcraft.yaml
  snapcraft init
3. I try to build the default snap:
  snapcraft --debug

Thereby the system produces the error commented below:

**To Reproduce**
How, and what happened?

```
ubuntu@ubuntu:~/snap-tests-BODAS$ snapcraft --debug
Launching a VM.
Build environment is in unknown state, cleaning first.
info failed: multipass socket access denied
Please check that you have read/write permissions to '/var/snap/multipass/common/multipass_socket'
launch failed: multipass socket access denied
Please check that you have read/write permissions to '/var/snap/multipass/common/multipass_socket'
An error occurred with the instance when trying to launch with 'multipass': returned exit code 2.
Ensure that 'multipass' is setup correctly and try again.

```

I tried to manually activate multipass.multipassd, but it gets deactivated after few seconds:

    ubuntu@ubuntu:~$ snap services multipass
    Service Startup Current Notes
    multipass.multipassd enabled inactive -
    ubuntu@ubuntu:~$ sudo snap restart multipass.multipassd
    Restarted.
    ubuntu@ubuntu:~$ snap services multipass
    Service Startup Current Notes
    multipass.multipassd enabled active -
    ubuntu@ubuntu:~$ snap services multipass
    Service Startup Current Notes
    multipass.multipassd enabled inactive -

so when I try ’ systemctl status snap.multipass.multipassd.service’, then I get:

grafik
[grafik963×197 10.4 KB](https://ubuntucommunity.s3.dualstack.us-east-2.amazonaws.com/original/2X/b/b286bf47d66b89b2131528082b86e07a58351975.png)

**Expected behavior**
What did you expect to happen?
I expect a correct execution of 'snapcraft --debug'

**Logs**
Please provide logs from the daemon, see [accessing logs](https://multipass.run/docs/accessing-logs) on where to find them on your platform.

![grafik](https://user-images.githubusercontent.com/64407895/174326858-74111052-9660-46d7-803d-dbec47147b2f.png)

**Additional info**
 - OS: ubuntu 20.04.4 LTS (GNU/Linux 5.4.0-1065-raspi armv7l)
- `multipass version`  1.9.2
- `multipass info --all`: info failed: multipass socket access denied

**Additional context**

**Comment by townsend2010** [comment]
Hi @iscopa,

So `multipassd` is failing when trying to set group ownership of the Multipass socket at `/var/snap/multipass/common/multipass_socket`.  This `chown()` operation is run under `root`, so it's a bit puzzling why this isn't being permitted on your system.

Just to see, you could try creating a file in `/var/snap/multipass/common` using the `root` user and then try to `chown root.sudo` this file and see if it succeeds.

**Comment by iscopa** [comment]
Thank you,

I have tried the following instructions but unfortunately that did not work either:

![grafik](https://user-images.githubusercontent.com/64407895/174591737-39295d39-d305-4ebd-aa32-4fbebc84c0e1.png)

**Comment by iskopa** [comment]
I am also trying to install snap on the raspberry pi 3 using another OS: The default Raspberry Pi OS 32bit.

I have carried out the steps indicated here:
    https://snapcraft.io/install/snapcraft/raspbian

Thereby I got the following error :
 
![image](https://user-images.githubusercontent.com/75910966/174623850-c183d52a-c34c-4135-917f-338829ff804d.png)

I am working with a Raspberry Pi 3 model B version 1.2, but these OS seem not to be compatible with snap.

Can you please recommend an OS for this device enabling snap installation and snap generation?

Thank you in advance and regards

**Comment by townsend2010** [comment]
Hi @iscopa,

My instructions weren't to fix the error, but instead was a way to help diagnose what your issue may be.  I don't have a Raspberry Pi, so I don't have a way to debug, so I'm asking for your help :wink: 

Could you please do the following:
`cd /var/snap/multipass/common`
`sudo touch test`
`$ ls -la`
Please provide the output of the previous command.
`$ sudo chown root.sudo test`
If that fails, please provide the output.  If it doesn't fail, please provide the output of:
`$ ls -la`

We'll see what this yields and go from there.

**Comment by townsend2010** [comment]
Great, could you please provide the output of `ls -la` after the `chown` command?  Thanks!

**Comment by iscopa** [comment]
sure:

![grafik](https://user-images.githubusercontent.com/64407895/174827523-57ea2fca-8674-49de-ace7-4e6956dfb14f.png)

**Comment by townsend2010** [comment]
Ok, so that works as expected.  I see `multipass_socket`  is in that directory, so let's try changing ownership by hand and see what happens.

`$ sudo chown root.sudo multipass_socket`

and then see if that works.  It's not the solution if it does, but it might help point us in the right direction.

**Comment by iscopa** [comment]
I executed the instruction above and tried to generate a snap:

![grafik](https://user-images.githubusercontent.com/64407895/174970936-34f308d0-84e7-4cdf-ad61-f2da3a4b910e.png)

**Comment by iscopa** [comment]
thank you. We could solve the snap build issue with this instruction

snapcraft clean && snapcraft --destructive-mode

**Comment by iscopa** [comment]
we changed the ownership of the app as well :

sudo chown -R ubuntu:ubuntu ./can-demo-python

Mit freundlichen Grüßen / Best regards

Isidro Corral

Cluster Distributed Intelligence (CR/ADI1.2)
Robert Bosch GmbH | Renningen | 70465 Stuttgart | GERMANY | www.bosch.com
Tel. +49 711 811-10228 | Mobile +49 1525 8813113 | ***@***.******@***.***>

Registered Office: Stuttgart, Registration Court: Amtsgericht Stuttgart, HRB 14000;
Chairman of the Supervisory Board: Prof. Dr. Stefan Asenkerschbaumer; Managing Directors: Dr. Stefan Hartung,
Dr. Christian Fischer, Filiz Albrecht, Dr. Markus Forschner, Dr. Markus Heyn, Rolf Najork
​
Von: Chris Townsend ***@***.***>
Gesendet: Mittwoch, 22. Juni 2022 15:03
An: canonical/multipass ***@***.***>
Cc: Corral Isidro (CR/ADI1.2) ***@***.***>; Mention ***@***.***>
Betreff: Re: [canonical/multipass] Multipass launch failed (Issue #2620)


Closed #2620<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fcanonical%2Fmultipass%2Fissues%2F2620&data=05%7C01%7Cisidro.corralpatino%40de.bosch.com%7C13364c7fae1e47e5ee1d08da544f90fe%7C0ae51e1907c84e4bbb6d648ee58410f4%7C0%7C0%7C637914997978673632%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=sof%2FyfTHgwzwzkFe1KwzvNyQEblq1dWL4LCkPPql1jc%3D&reserved=0> as completed.
